src/components/List.jsx

```diff
@@ -0,0 +1,129 @@
+import styled from "styled-components";
+import TodoItem from "./TodoItem";
+import { useState, useMemo } from "react";
+
+const List = ({ todos, onUpdate, setTodos, removeTodo }) => {
+  const [search, setSearch] = useState("");
+
+  const onChangeSearch = (e) => {
+    setSearch(e.target.value);
+  };
+
+  const getFilteredDate = () => {
+    if (search === "") {
+      return todos;
+    }
+    return todos.filter((todos) => {
+      return todos.content.toLowerCase().includes(search.toLowerCase());
+    });
+  };
+
+  const filteredTodos = getFilteredDate();
+
+  const { doneCount, notDoneCount } = useMemo(() => {
+    const totalCount = todos.length;
+    const doneCount = todos.filter((todo) => todo.isDone).length;
+    const notDoneCount = totalCount - doneCount;
+
+    return {
+      totalCount,
+      doneCount,
+      notDoneCount,
+    };
+  }, [todos]);
+
+  return (
+    <ListContainer>
+      <TopWrapper>
+        <h4>Todo List 🌱</h4>
+        <CountWrapper>
+          <span>todo : {notDoneCount}</span>
+          <span>done : {doneCount}</span>
+        </CountWrapper>
+      </TopWrapper>
+      <SearchInput
+        type="text"
+        placeholder="검색어를 입력하세요"
+        value={search}
+        onChange={onChangeSearch}
+      />
+      <TodoWrapper>
+        {filteredTodos.map((todos) => {
+          return (
+            <TodoItem
+              key={todos.id}
+              {...todos}
+              onUpdate={onUpdate}
+              todos={todos}
+              setTodos={setTodos}
+              removeTodo={removeTodo}
+            />
```

Comment:
저도 같은 가빈 님과 같은 생각입니다! 지금 setTodos를 props로 내보내고 있는데, 실제 TodoItem 컴포넌트에서 setTodos를 사용하지 않고 있어요. 불필요한 props는 제거하는 게 좋을 것 같습니다.

그리고 {...todos} 부분과  todos={todos} 부분도 중복된 것 같습니다. 이미 {...todos}로 todos 안의 모든 속성들을 전달하고 있기 때문에 굳이 todos={todos}로 두 번 전달할 필요가 없어요..! 이 부분도 삭제하면 더욱 깔끔한 코드가 될 것 같습니다 🙃

```suggestion
            <TodoItem
              key={todos.id}
              {...todos}
              onUpdate={onUpdate}
              removeTodo={removeTodo}
            />
```